### PR TITLE
Refactored the fetch requests to use a client for Blizzard's API

### PR DESCRIPTION
**Pull request by jteague9**
Also added legacy endpoint support.

### CODE REVIEW

File: app/src/update_db.py [hunk 1]
```diff
@@ -3,15 +3,24 @@
...
 
 
-def update_all_for_region(region):
-    loop = asyncio.new_event_loop()
-    asyncio.set_event_loop(loop)
-    future = asyncio.ensure_future(get_all_ladder_responses_for_region(region))
-    ladder_list = loop.run_until_complete(future)
+async def fetch_ladders(region=None):
+    loop = asyncio.get_event_loop()
+    blz = BlizzSession(os.environ['BLIZZARD_CLIENT_ID'], os.environ['BLIZZARD_CLIENT_SECRET'], loop=loop)
```
Comment (tso) [inline app/src/update_db.py]: use `settings.BLIZZARD_CLIENT_ID` instead

File: app/src/parse.py [hunk 2]
```diff
@@ -1,38 +1,72 @@
...
-    for i in response_json.get('team'):
-        for j in i.get('member'):
-            for k in j.get('played_race_count'):
-                realm = j.get('legacy_link', {}).get('realm')
-                race = k['race']['en_US']
-                character_name = j.get('legacy_link', {}).get('name')
-                bnet_id = j.get('character_link', {}).get('battle_tag')
-                profile_id = j.get('character_link', {}).get('id')
-                clan_tag = j.get('clan_link', {}).get('clan_tag')
+    rank = int(response_json.get('league', {}).get('league_key', {}).get('league_id'))
```
Comment (tso) [inline app/src/parse.py]: can we change this to just `int(response_json['league']['league_key']['league_id']`? It'll throw when we try to Int `None` anyway right?

File: app/src/parse.py [hunk 3]
```diff
@@ -1,38 +1,72 @@
...
+    for team in response_json.get('team', []):
+        for member in team.get('member'):
```
Comment (tso) [inline app/src/parse.py]: Will throw if we try to iterate over None so just `team['member']`?

File: app/src/parse.py [hunk 4]
```diff
@@ -1,38 +1,72 @@
...
+            for k in member.get('played_race_count'):
```
Comment (tso) [inline app/src/parse.py]: Will throw if we try to iterate over None so just `member['place_race_count']`?

File: app/src/parse.py [hunk 5]
```diff
@@ -1,38 +1,72 @@
...
+                    bnet_id = member['character_link']['battle_tag']
+                    profile_id = member['character_link']['id']
+                    clan_tag = member.get('clan_link', {}).get('clan_tag')
+                    mmr = team['rating']
+                    wins = team['wins']
+                    losses = team['losses']
                     ladder_list.append(Player.players.create_player(
                         realm=realm,
                         region=region.upper(),
                         rank=int(rank),
```
Comment (tso) [inline app/src/parse.py]: rank is already inted so just `rank=rank`

File: app/src/update_db.py [hunk 6]
```diff
@@ -20,13 +29,49 @@ def update_all_for_region(region):
...
+
+def update_gm_for_region_legacy(region):
+    loop = asyncio.new_event_loop()
+    data = loop.run_until_complete(fetch_gm_legacy(region))
+    update_database_legacy(region, [data])
+
+
+async def fetch_gm_legacy(region):
+    loop = asyncio.get_event_loop()
+    blz = BlizzSession(os.environ['BLIZZARD_CLIENT_ID'], os.environ['BLIZZARD_CLIENT_SECRET'], loop=loop)
```
Comment (tso) [inline app/src/update_db.py]: `settings.BLIZZARD_CLIENT_ID`

File: app/src/update_db.py [hunk 7]
```diff
@@ -20,13 +29,49 @@ def update_all_for_region(region):
...
-    update_all_for_region(Region.KR.value.lower())
+    loop = asyncio.new_event_loop()
+    data = loop.run_until_complete(fetch_ladders())
+    loop.close()
+    for region in data:
+        update_database(region['region'], region['data'])
+
+
+def update_all_gm_legacy():
+    for region in [Region.US.value.lower(), Region.EU.value.lower(), Region.KR.value.lower()]:
```
Comment (tso) [inline app/src/update_db.py]:
```
for _, region in Region.choices():
  update_gm_for_region_legacy(region.lower())
```

**Review by tso — APPROVED** [review]
nice

**Review by WilliamStewart — APPROVED** [review]

File: app/src/update_db.py (hunk 6 above)
Comment (jteague9) [reply app/src/update_db.py]: Updated in 23419a6

**Comment by jteague9** [comment]
Updated all suggestions in 23419a6

**Pull request merged**